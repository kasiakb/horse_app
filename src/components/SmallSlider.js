import 'rc-slider/assets/index.css';
import 'rc-tooltip/assets/bootstrap.css';
import React from 'react';
import Slider from 'rc-slider';
import Tooltip from 'rc-tooltip';

const Handle = Slider.Handle;

class SmallSlider extends React.Component {
  wrapperStyle = { width: 400, margin: 50 };

  handleSliderChange = (defaultValue) => {
    this.props.passChange(defaultValue, this.props.title)
  }

  componentWillReceiveProps(nextProps) {
    this.forceUpdate()
  }

  render() {  
    console.log('props', this.props)
    return (
      <div>
        <form>
          <div style={ {width: 400, margin: 50} } key={this.props.title}> 
            <label>
              {this.props.title}
              <Slider 
                className={this.props.title}
                name={this.props.title} 
                min={this.props.minCapacity} 
                max={this.props.maxCapacity} 
                step={0.01} 
                defaultValue={this.props.defaultValue}
                handle={this.props.handle} 
                onChange={this.handleSliderChange}
                marks = {this.props.marks}
                value = {this.props.defaultValue}
                />
            </label>
          </div>
        </form>
      </div>
    );
  }
}

export default SmallSlider