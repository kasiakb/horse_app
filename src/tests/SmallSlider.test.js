import renderer from 'react-test-renderer';
import SmallSlider from '../components/SmallSlider';
import React from 'react';
import ReactDOM from 'react-dom';
import Slider from 'rc-slider'

describe('SmallSlider component renders the small sliders correctly', () => {
  it('renders correctly', () => {
    const key = "Owies"
    const title = "Owies"
    const marks={
      0: "0 kg",
      2: "2 kg"}
    const handle = jest.fn()
    const passChange = jest.fn()
    
    const rendered = renderer.create(
      <SmallSlider
      key={key}
      title={title}
      defaultValue={0}
      marks={marks}
      minCapacity={0}
      maxCapacity={2}
      handle={handle}
      passChange={passChange}
    />
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});