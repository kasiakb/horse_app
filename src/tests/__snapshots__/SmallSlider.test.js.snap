// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmallSlider component renders the small sliders correctly renders correctly 1`] = `
<div>
  <form>
    <div
      style={
        Object {
          "margin": 50,
          "width": 400,
        }
      }
    >
      <label>
        Owies
        <div
          className="rc-slider rc-slider-with-marks Owies"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          onTouchStart={[Function]}
          style={undefined}
        >
          <div
            className="rc-slider-rail"
            style={Object {}}
          />
          <div
            className="rc-slider-track"
            style={
              Object {
                "left": "0%",
                "visibility": "visible",
                "width": "0%",
              }
            }
          />
          <div
            className="rc-slider-step"
          >
            <span
              className="rc-slider-dot rc-slider-dot-active"
              style={
                Object {
                  "left": "0%",
                }
              }
            />
            <span
              className="rc-slider-dot"
              style={
                Object {
                  "left": "100%",
                }
              }
            />
          </div>
          <div
            className="rc-slider-mark"
          >
            <span
              className="rc-slider-mark-text rc-slider-mark-text-active"
              style={
                Object {
                  "left": "0%",
                  "marginLeft": "-45%",
                  "width": "90%",
                }
              }
            >
              0 kg
            </span>
            <span
              className="rc-slider-mark-text"
              style={
                Object {
                  "left": "100%",
                  "marginLeft": "-45%",
                  "width": "90%",
                }
              }
            >
              2 kg
            </span>
          </div>
        </div>
      </label>
    </div>
  </form>
</div>
`;
